File: AspNetMvcUnleashed/0672329980_source/Chapter03/VB/MvcApplication1/MvcApplication1/Controllers/ContentManagerController.vb
Public Class ContentManagerController
    Inherits System.Web.Mvc.Controller


    Function Index()
        Return View()
    End Function

    Function Download() As ActionResult
        Return File("~/Content/CompanyPlans.docx", "application/vnd.openxmlformats-officedocument.wordprocessingml.document", "CompanyPlans.docx")
    End Function

End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter03/CS/MvcApplication1/MvcApplication1/Controllers/ContentManagerController.cs
using System.Web.Mvc;

namespace MvcApplication1.Controllers
{
    public class ContentManagerController : Controller
    {
 
        public ActionResult Index()
        {
            return View();
        }


        public ActionResult Download()
        {
           
            return File("~/Content/CompanyPlans.docx", "application/vnd.openxmlformats-officedocument.wordprocessingml.document", "CompanyPlans.docx");
        }

    }
}
